db/db_impl/db_impl.cc

```diff
@@ -6073,11 +6077,8 @@ Status DBImpl::IngestExternalFiles(
       autovector<autovector<VersionEdit*>> edit_lists;
       uint32_t num_entries = 0;
       for (size_t i = 0; i != num_cfs; ++i) {
-        auto* cfd =
-            static_cast<ColumnFamilyHandleImpl*>(args[i].column_family)->cfd();
-        if (cfd->IsDropped()) {
-          continue;
-        }
+        auto* cfd = ingestion_jobs[i].GetColumnFamilyData();
+        assert(!cfd->IsDropped());
```

Comment:
I need to come back to verify this is okay but could you help point out what is the existing assertion for it?

Reply:
Yeah, so here is where column family can be dropped: https://github.com/facebook/rocksdb/blob/1006eddd635539466d6d99a3396de3c1df4e1993/db/db_impl/db_impl.cc#L3728-L3733

It only starts after entering the single threaded write queue. And the column family's dropped bit is only flagged when this VersionEdit is written successfully to Manifest file: https://github.com/facebook/rocksdb/blob/1006eddd635539466d6d99a3396de3c1df4e1993/db/version_set.cc#L5722

In file ingestion, we first enter the write queue, and does a column family dropped check here: https://github.com/facebook/rocksdb/blob/1006eddd635539466d6d99a3396de3c1df4e1993/db/db_impl/db_impl.cc#L6018

This would mean if we don't find a dropped column family among the specified column families at this point, we shouldn't see newly dropped column families before we exit the single threaded write queue. If there is newly dropped column family, I think that should be asserted and worth investigating.

Reply:
There is no existing assertion for it. The existing logic will ignore such a case. But the new Memtables status check I added need to take into consideration what happens if the column family is dropped, since flushing logic will not attempt to really flush dropped column families. To simplify that logic, I instead add assertion here that as long as the column family gets past that first dropped check, follow up logic in file ingestion should not be concerned about the possibility of column families being dropped until we exit the write thread.